src/azure.rs

```diff
@@ -287,95 +292,48 @@ impl ObjectStore for MicrosoftAzure {
     }
 
     async fn list(&self, prefix: Option<&Path>) -> Result<BoxStream<'_, Result<ObjectMeta>>> {
-        #[derive(Clone)]
```

Comment:
This is a drive-by-fix, the listing logic is extracted into a `list_impl` so that it can be used by `list_with_delimiter`